### ISSUE TITLE
Write unit tests for protocol registration 

### ISSUE THREAD

**Issue by abstralexis** [ResearchPocket/ResearchPocket#16]
_(Parent Issue: #6)_
This may be something to implement in more places than just the protocol registration, however testing here would be quite beneficial so users aren't confused as to why something doesn't work. Optionally, GitHub Actions can be set up to cargo test as a requirement before releases.

**Comment by KorigamiK** [comment]
Yes for sure. We definitely lack testing for a lot if things so contributions are welcome 🤗
